### PR TITLE
deps: update osbuild/images to v0.91.0

### PR DESCRIPTION
**Pull request by achilleas-k**
- Update osbuild/images dependency
- Update Go version to v1.22
- Update golangci-lint version v1.61.0 (for Go v1.22)

The updated linter detects a few new issues, which have now been fixed or annotated to be ignored.

### CODE REVIEW

File: cmd/osbuild-worker/jobimpl-koji-finalize.go [hunk 1]
```diff
@@ -305,13 +311,16 @@ func (impl *KojiFinalizeJobImpl) Run(job worker.Job) error {
 		}
 	}
 
+	if args.StartTime > math.MaxInt64 {
```
Comment (schuellerf) [inline cmd/osbuild-worker/jobimpl-koji-finalize.go:314]: when would that happen? StartTime _is_ uint64?

File: internal/cloudapi/v2/handler.go [hunk 2]
```diff
@@ -448,7 +453,11 @@ func (h *apiHandlers) getComposeStatusImpl(ctx echo.Context, id string) error {
 			ImageStatuses: &buildJobStatuses,
 			KojiStatus:    &KojiStatus{},
 		}
-		buildID := int(initResult.BuildID)
+
+		if initResult.BuildID < math.MaxInt {
```
Comment (schuellerf) [inline internal/cloudapi/v2/handler.go:457]: will this ever be false?

**Review by schuellerf — CHANGES_REQUESTED** [review]
Those range checks seem strange or is there a reason for that?
(also for the new if-branches some tests would be missing, if I would nitpick …)

File: cmd/osbuild-worker/jobimpl-koji-finalize.go (hunk 1 above)
Comment (achilleas-k) [reply cmd/osbuild-worker/jobimpl-koji-finalize.go:314]: Right, but it's comparing with max *int* 64 (not u).

File: internal/cloudapi/v2/handler.go (hunk 2 above)
Comment (achilleas-k) [reply internal/cloudapi/v2/handler.go:457]: My mistake.  That check should be `< 0`.

File: cmd/osbuild-worker/jobimpl-koji-finalize.go (hunk 1 above)
Comment (schuellerf) [reply cmd/osbuild-worker/jobimpl-koji-finalize.go:314]: You are totally right - I overlooked that :-/

File: cmd/osbuild-worker/jobimpl-koji-finalize.go [hunk 3]
```diff
@@ -102,6 +103,11 @@ func (impl *KojiFinalizeJobImpl) Run(job worker.Job) error {
 		// Fail the Koji build if the job error is set and the necessary
 		// information to identify the job are available.
 		if kojiFinalizeJobResult.JobError != nil && initArgs != nil {
+			if initArgs.BuildID > math.MaxInt {
+				logWithId.Errorf("job BuildID %d can't be converted to 'int'", initArgs.BuildID)
```
[the remark this answers is not in the document]
Comment (schuellerf) [reply cmd/osbuild-worker/jobimpl-koji-finalize.go:107]: same oversight on my end here

[the remark this answers is not in the document]
Comment (achilleas-k) [reply cmd/osbuild-worker/jobimpl-koji-finalize.go:107]: That's a good question.  I'm not entirely sure either.  I (incorrectly) reflexively read this as checking if BuildID is > `MaxUint64`.  I'm not sure if we ever considered what would happen if osbuild-composer was built for (or ran on) a 32-bit system.

**Comment by achilleas-k** [comment]
Looks like RHEL 9.4 is unhappy (Go version issue).

**Pull request closed without merging**